src/main/java/org/opensearch/knn/common/KNNConstants.java

```diff
@@ -115,4 +115,6 @@ public class KNNConstants {
     // Please refer this github issue for more details for choosing this value:
     // https://github.com/opensearch-project/k-NN/issues/1049#issuecomment-1694741092
     public static int MAX_DISTANCE_COMPUTATIONS = 2048000;
+    public static int MAX_ID_SELECT_ARRAY = 2048;
```

Comment:
This value is too small. What is the reason for choosing this value? Last I remember we can have 10^6 array size.

Reply:
We need to change this, I would like to think more on this before hard coding this value. See faiss_wrapper code to identify how we are using count of ids to predict the size of the array and bitmap and take a call on what to use. That is better than the hard coded values.

Reply:
> This is the 1st version for the array size. we can change it.

what do you mean by we can change it? Are you thinking to merge this code in a feature branch? I was thinking to merge the code in main branch directly.